libdnf/dnf-context.cpp

```diff
@@ -2498,3 +2510,30 @@ pluginGetContext(DnfPluginInitData * data)
     }
     return (static_cast<PluginHookContextInitData *>(data)->context);
 }
+
+namespace libdnf {
+
+std::map<std::string, std::string> &
+dnf_context_get_vars(DnfContext * context)
+{
+    return *GET_PRIVATE(context)->vars;
+}
+
+bool
+dnf_context_get_vars_cached(DnfContext * context)
+{
+    return GET_PRIVATE(context)->varsCached;
+}
+
+void
+dnf_context_load_vars(DnfContext * context)
+{
+    auto priv = GET_PRIVATE(context);
+    priv->vars->clear();
+    ConfigMain::addVarsFromEnv(*priv->vars);
+    for (auto dir = priv->vars_dir; *dir; ++dir)
+        ConfigMain::addVarsFromDir(*priv->vars, std::string(priv->install_root) + *dir);
```

Comment:
Shouldn't this be:

```
    for (auto dir = priv->vars_dir; *dir; ++dir)
        ConfigMain::addVarsFromDir(*priv->vars, std::string(priv->source_root) + *dir);
    for (auto dir = priv->vars_dir; *dir; ++dir)
        ConfigMain::addVarsFromDir(*priv->vars, std::string(priv->install_root) + *dir);
     ConfigMain::addVarsFromEnv(*priv->vars);
```

This change would cover the following use cases:
- I'm using DNF to bootstrap an installation (aka anaconda or `dnf --installroot`) as I may have vars defined on the running system that are not in the install root.  In this layout `install_root` variables would override the `source_root` variables
- I want to alter DNF's behavior via environment variables.  With the code you've got, if you try to redefine a var that is on the filesystem, your env var will be ignored.